File: server/src/CoinyProject.Application/Features/Albums/Models/AlbumModel.cs
using CoinyProject.Application.Common.Models;
using CoinyProject.Domain.Enums;

namespace CoinyProject.Application.Features.Albums.Models;

public class AlbumModel
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int Rate { get; set; }
    public AlbumStatus Status { get; set; }
    public BaseLink Author { get; set; }
    public AlbumElementLink[] Images { get; set; }

    public DateTime UpdatedAt { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: client/src/app/features/albums/models/album.model.ts
import {AlbumStatusEnum} from "@core/models/enums/album-status.enum";
import {BaseLinkModel} from "@shared/models/base-link.model";
import {AlbumElementLinkModel} from "@shared/models/album-element-link.model";

export interface AlbumGetModel {
  id: string;
  name: string;
  description: string;
  rate: number;
  status: AlbumStatusEnum;
  author: BaseLinkModel;
  images: AlbumElementLinkModel[];
  updatedAt: string;
  createdAt: string;
}
